week2/homework/src/app.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+let program = require('commander');
+const {
+  promisify
+} = require('util');
+
+const {
+  readFile,
+  writeFile,
+  appendFile
+} = require('fs');
+const TODO_FILE = 'todo.json';
+
+
+const readFileWithPromise = promisify(readFile);
+const writeFileWithPromise = promisify(writeFile);
+const appendFileWithPromise = promisify(appendFile);
+
+
+program
+  .version('0.1.0')
+  .option('-a, --add', 'Add TODO')
+  .option('-l, --list', 'List TODOs')
+  .option('-r, --remove', 'Remove specific TODO')
+  .option('-d, --reset', 'Remove All TODOs')
+  .option('-u, --update', 'Update specific TODO')
+  .parse(process.argv);
+
+async function main() {
+  const [, , cmd, ...args] = process.argv;
+
+  if (program.add) {
+    const data = await readFileWithPromise(TODO_FILE, 'utf-8').catch(() => '[]');
+    const todos = JSON.parse(data);
+    const newTodo = args.join(' ');
+    todos.push(newTodo);
+    console.log(newTodo);
+    await writeFileWithPromise(TODO_FILE, JSON.stringify(todos));
+
+  }
+  if (program.list) {
+    const data = await readFileWithPromise(TODO_FILE, 'utf-8').catch(() => '[]');
+    const todos = JSON.parse(data);
+    console.info(todos);
+  }
+  if (program.remove) {
+    const data = await readFileWithPromise(TODO_FILE, 'utf-8').catch(() => '[]');
+    const todos = JSON.parse(data);
```

Comment:
i need help in this point !!

Reply:
?? any idea ?

Reply:
A function could do that for you right. If you have a function which does this: 
```js
const data = await readFileWithPromise(TODO_FILE, 'utf-8').catch(() => '[]');
const todos = JSON.parse(data);
```
then you just have to execute that function when you need it. Let's say you call it `getTodos`, then instead of writing those two lines here, you can just say `getTodos()`. It doesn't change anything to the logic of you code per se, but it makes your code cleaner, more readable, shorter and ensures that you get your todos (because this is actually what you want) in the same way each time. Additionally, if later on you want to get the todos in a different way, you'll only have to change your code once inside of the `getTodos` function.

Does that make sense?